x/ibc/core/02-client/types/msgs.go

```diff
@@ -75,17 +74,20 @@ func (msg MsgCreateClient) ValidateBasic() error {
 	if err := clientState.Validate(); err != nil {
 		return err
 	}
-	if clientState.ClientType() == exported.Localhost || msg.ClientId == exported.Localhost {
+	if clientState.ClientType() == exported.Localhost {
 		return sdkerrors.Wrap(ErrInvalidClient, "localhost client can only be created on chain initialization")
 	}
 	consensusState, err := UnpackConsensusState(msg.ConsensusState)
 	if err != nil {
 		return err
 	}
-	if err := consensusState.ValidateBasic(); err != nil {
-		return err
+	if clientState.ClientType() != consensusState.ClientType() {
+		return sdkerrors.Wrap(ErrInvalidClientType, "client type for client state and consensus state do not match")
 	}
-	return host.ClientIdentifierValidator(msg.ClientId)
+	if err := ValidateClientType(clientState.ClientType()); err != nil {
+		return sdkerrors.Wrap(err, "client type does not meet naming constraints")
```

Comment:
needs test

Reply:
I think this would require implementing a mock light client